### ISSUE TITLE
Bad naming convention for service name

### ISSUE THREAD

**Issue by lujiajing1126** [jaegertracing/jaeger-operator#1360]
Currently the naming convention for the collector service is determined in this function,
https://github.com/jaegertracing/jaeger-operator/blob/010535a2ac314a0106ae0ac31ea136b85dcb7f26/pkg/service/collector.go#L97

which results in some name like `grpc-http(s)`.

However, this will cause confusing for istio as documented in the [`protocol-selection`](https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/) section, from which I excerpt as follows,

- By the name of the port: `name: <protocol>[-<suffix>]`.
- In Kubernetes 1.18+, by the `appProtocol` field: `appProtocol: <protocol>`.

The result of this bad name is severe, the istio-sidecar will automatically convert grpc protocol to http one which cannot be correctly processed by `collector` service.

Thus, I suggest changing the naming convention to follow the istio rules. I would like to submit a PR to resolve this issue.

**Comment by jpkrohling** [comment]
Could you provide a test for the case you described? In any case, a PR would certainly be appreciated.

**Comment by lujiajing1126** [comment]
Attached there are two captured tcpdump logs for the scenarios with or without istio-sidecar injected to the application pod,
(while in the same time, we do not inject istio-sidecar for jaeger-collector  pod by default)

- [jaeger-agent-with-istio-sidecar.log](https://github.com/jaegertracing/jaeger-operator/files/5801124/jaeger-agent-with-istio-sidecar.log)
- [jaeger-agent-without-istio-sidecar.log](https://github.com/jaegertracing/jaeger-operator/files/5801125/jaeger-agent-without-istio-sidecar.log)

As we can see, if the istio-sidecar is injected, the grpc stream is transcoded to http request automatically.

I suppose I can use this function to reproduce the issue? Or we need a full example?

https://github.com/istio/istio/blob/68ebdc1f1eb9fe832e6ddc26962c0630f9275de1/pkg/config/kube/conversion.go#L47-L79

**Comment by jpkrohling** [comment]
So, this is a problem only for pods with both Jaeger and Istio sidecars? @jkandasa do we have tests for this scenario?

**Comment by lujiajing1126** [comment]
> So, this is a problem only for pods with both Jaeger and Istio sidecars? @jkandasa do we have tests for this scenario?

Exactly.  Specifically,

```
| Pod: Application --(spans)--> Jaeger Agent --(network intercepted by)--> Istio Sidecar | 
                    |
                    |
                    |
                    |
                    |
                    |
                    v
| Pod: (without Istio Sidecar) Jaeger Collector |
```

I haven't tested the case with both istio sidecar injected.

**Comment by jpkrohling** [comment]
Was this closed as part of #1368?

**Comment by lujiajing1126** [comment]
> Was this closed as part of #1368?

It should have been closed. I don't know what happened to the Mergify bot.

**Comment by Stevmann** [comment]
Hello, 

I have an issue described in #1122 with port name set to "grcp-https": 

"authentication handshake failed: tls: first record does not look like a TLS handshake"

If I set the port name to string which was changed in this issue; "https-grpc" then it works. 

---
jaeger-operator.v1.29.1
OpenShift 4.6.43
Istio version OSSM_2.1.1-1.el8
